src/components/Contact.tsx

```diff
@@ -1,38 +1,43 @@
-import React, { useState } from 'react'
 import axios from 'axios'
+import { FieldValues, useForm } from 'react-hook-form'
 import InputField from './InputField'
-import Button from './Button'
-import { FormContainer, TextContainer } from './styles/FormContainer.styled'
-import { contactUs } from '../texts/descriptions/contactUs'
+import SubmitButton from './SubmitButton'
+import { emailRegex, nameRegex } from '../texts/errors/formErrors'
 
-export function ContactForm() {
-  const [cName, setName] = useState<string>('')
-  const [cEmail, setEmail] = useState<string>('')
-  const [cMessage, setMessage] = useState<string>('')
-  const onSubmit = () => {
+import { FormContainer } from './styles/FormContainer.styled'
+
+function ContactForm() {
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ mode: 'onChange' })
+
+  const onSubmit = (data: FieldValues) => {
     axios({
       url: 'https://formspree.io/f/mvolznvk', // placeholder, replace with ur id here
       method: 'post',
       headers: {
         Accept: 'application/json',
       },
       data: {
-        name: cName,
-        email: cEmail,
-        message: cMessage,
+        name: data.Name.trim().replace(/ {2,}/g, ' '), // trim and remove double spaces
+        email: data.Email.trim(),
+        message: data.Message.trim(),
       },
     }).then((response) => {
       console.log(response)
     })
   }
 
   return (
-    <FormContainer>
-      <TextContainer>{contactUs}</TextContainer>
-      <InputField title="Name" value={cName} setValue={setName} type="name" required />
-      <InputField title="Email" value={cEmail} setValue={setEmail} type="email" required />
-      <InputField title="Message" value={cMessage} setValue={setMessage} type="text" required />
-      <Button text="Send" onClick={onSubmit} fontSize="30px" />
+    <FormContainer onSubmit={handleSubmit(onSubmit)} noValidate>
+      <InputField type="text" title="Name" errors={errors} register={register} pattern={nameRegex} required />
+      <InputField type="email" title="Email" errors={errors} register={register} pattern={emailRegex} required />
+      <InputField type="text" title="Message" errors={errors} register={register} required />
```

Comment:
would it be better to pass in the exact error pertaining to this input? ie errors.Message?

Reply:
changed